dlib/math/details/bessel.h

```diff
@@ -0,0 +1,470 @@
+// Copyright (C) 2006  Davis E. King (davis@dlib.net)
+// License: Boost Software License   See LICENSE.txt for the full license.
+#ifndef DLIB_MATH_DETAIL_BESSEL
+#define DLIB_MATH_DETAIL_BESSEL
+
+#include <cmath>
+#include <limits>
+#include <stdexcept>
+
+namespace dlib
+{
+    namespace detail
+    {
+        template <typename R>
+        void bessel_ik(R nu, R x, R& Inu, R& Knu, R& Ipnu, R& Kpnu)
```

Comment:
Where is this code from?  It's got some license to it presumably.

Reply:
It's pretty much copy pasted from libstdc++

Reply:
Hmm, the software license of libstdc++ is GPL. Should i copy-paste from Boost instead? That has friendlier license.

Reply:
Yeah definitely don't copy stuff from other libraries unless those also use the boost software license.  And then in that case also preserve the copyright notice on it.

Although you are using this stuff right?  And it's unreasonable to use boost or c++17 to get the functions?

Reply:
It is reasonable to use those but not if i want stuff self-contained in dlib. I want to add STFT functions to dlib. That requires window functions, which require bessel functions. Ideally I would like to do STFTs in dlib without having to add C++17 or boost as firm dependencies to dlib.

Reply:
Weirdly, i don't think libc++ supports special math functions yet. They are 5 years behind

Reply:
> Also, if you so much as breath on 1 header file in boost, you've accidentally included 42000 other headers and now need to link to at least 1 boost shared/static library.

Ha, yeah I know this pain of boost too 😅